test/e2e/utils.ts

```diff
@@ -26,24 +28,72 @@ export const createLocalAthensDB = async (page:Page, dbName:string) => {
     await page.pause()
 };
 
+export const lastBlockSelector = '.textarea >> nth=-1';
+
+export const inputInLastBlock = async (page:Page, text:string) => {
+    await page.click(lastBlockSelector);
+    await page.fill(lastBlockSelector, text);
+};
+
 export const saveLastBlock = async (page:Page, text:string) => {
-    await page.click('.textarea >> nth=-1');
-    await page.fill('.textarea >> nth=-1', text);
-    return page.press('.textarea >> nth=-1', 'ArrowUp');
+    await inputInLastBlock(page, text);
+    // Move away from the block to force a save.
+    return page.press(lastBlockSelector, 'ArrowUp');
 };
 
 export const saveLastBlockAndEnter = async (page:Page, text:string) => {
-    await page.click('.textarea >> nth=-1');
-    await page.fill('.textarea >> nth=-1', text);
-    return page.press('.textarea  >> nth=-1', 'Enter');
+    await inputInLastBlock(page, text);
+    await page.press(lastBlockSelector, 'Enter');
+    // Wait a bit for transaction and focus events to be resolved between saves.
+    // Without waiting, it's possible for multiple calls to end up in the same block instead of
+    // different blocks, because by the time the second call runs the new block hasn't appeared yet.
+    // TODO: we shouldn't need to do this, instead we should have deterministic states from input.
+    await page.waitForTimeout(200);
 };
 
 export const indentLastBlock = async (page:Page) => {
-    await page.click('.textarea >> nth=-1');
-    return page.press('.textarea  >> nth=-1', 'Tab');
+    await page.click(lastBlockSelector);
+    return page.press(lastBlockSelector, 'Tab');
 };
 
 export const unindentLastBlock = async (page:Page) => {
-    await page.click('.textarea >> nth=-1');
-    return page.press('.textarea  >> nth=-1', 'Shift+Tab');
+    await page.click(lastBlockSelector);
+    return page.press(lastBlockSelector, 'Shift+Tab');
 };
+
+
+export const goToDailyPages = async (page:Page) => {
+    // The sixth button is the daily notes button.
+    // TODO: find a better way to address this button, maybe tooltip?
```

Comment:
I think we could CSS classes to the Typescript components and select them that way.

Reply:
Usually for e2e we're looking at things a user could see in some way. A CSS class doesn't fit the bill, but a particular icon, text, or tooltip text does.

Reply:
Great explanation!